engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/SearchPredicateFactory.java

```diff
@@ -82,6 +84,64 @@ public interface SearchPredicateFactory {
 	@Deprecated
 	PredicateFinalStep bool(Consumer<? super BooleanPredicateClausesStep<?>> clauseContributor);
 
+	/**
+	 * Match documents if they match a conjunction of previously-built {@link SearchPredicate}.
+	 *
+	 * @return The initial step of a DSL where the "AND" predicate can be defined.
+	 * @see AndPredicateClausesStep
+	 */
+	AndPredicateClausesStep<?> and(
+			SearchPredicate firstSearchPredicate,
+			SearchPredicate... otherSearchPredicates);
+
+	/**
+	 * Match documents if they match a conjunction of boolean clauses.
+	 *
+	 * @return The initial step of a DSL where the "OR" predicate can be defined.
+	 * @see AndPredicateClausesStep
+	 */
+	AndPredicateClausesStep<?> and(PredicateFinalStep firstSearchPredicate,
+			PredicateFinalStep... otherSearchPredicates);
+
+	/**
+	 * Match documents if they match a conjunction of boolean clauses.
+	 *
+	 * @return The initial step of a DSL where the "AND" predicate can be defined.
+	 * @see AndPredicateClausesStep
+	 */
+	@SuppressWarnings("unchecked")
+	AndPredicateClausesStep<?> and(
+			Function<? super SearchPredicateFactory, ? extends PredicateFinalStep> firstSearchPredicate,
+			Function<? super SearchPredicateFactory, ? extends PredicateFinalStep>... otherSearchPredicates);
+
```

Comment:
I think this overload is worth the trouble.

Passing a function is an exotic use case to begin with (even with `.bool()`), and here it's also problematic because of the generics + varargs.

Can you remove this overload?

We can always add an `add(Function<....>)` method in the builder variant of `and()`/`or()`.